Source/WebKit/WebProcess/GPU/graphics/RemoteDisplayListRecorderProxy.cpp

```diff
@@ -70,22 +71,35 @@ RemoteDisplayListRecorderProxy::RemoteDisplayListRecorderProxy(RemoteRenderingBa
 template<typename T>
 ALWAYS_INLINE void RemoteDisplayListRecorderProxy::send(T&& message)
 {
-    auto imageBuffer = m_imageBuffer.get();
-    if (UNLIKELY(!m_renderingBackend))
+    RefPtr connection = this->connection();
+    if (UNLIKELY(!connection))
         return;
 
-    if (imageBuffer)
+    RefPtr imageBuffer = m_imageBuffer.get();
+    if (LIKELY(imageBuffer))
         imageBuffer->backingStoreWillChange();
-    auto result = m_renderingBackend->streamConnection().send(std::forward<T>(message), m_destinationBufferIdentifier, RemoteRenderingBackendProxy::defaultTimeout);
-#if !RELEASE_LOG_DISABLED
+    auto result = connection->send(std::forward<T>(message), m_destinationBufferIdentifier, RemoteRenderingBackendProxy::defaultTimeout);
     if (UNLIKELY(result != IPC::Error::NoError)) {
-        auto renderingBackendIdentifier  = m_renderingBackend->renderingBackendIdentifier();
-        RELEASE_LOG(RemoteLayerBuffers, "[renderingBackend=%" PRIu64 "] RemoteDisplayListRecorderProxy::send - failed, name:%" PUBLIC_LOG_STRING ", error:%" PUBLIC_LOG_STRING,
-            renderingBackendIdentifier.toUInt64(), IPC::description(T::name()).characters(), IPC::errorAsString(result).characters());
+        RELEASE_LOG(RemoteLayerBuffers, "RemoteDisplayListRecorderProxy::send - failed, name:%" PUBLIC_LOG_STRING ", error:%" PUBLIC_LOG_STRING,
+            IPC::description(T::name()).characters(), IPC::errorAsString(result).characters());
+        didBecomeUnresponsive();
```

Comment:
Should we only call this for `Error::Timeout`? I guess they're all errors that mean we can't really trust the state of the GPU process and we should probably kill it eagerly anyway (instead of then waiting for some subsequent timeout). But the name `didBecomeUnresponsive` doesn't match the other error types.

(And similar for the other rendering-related connection client classes we're adding this handling to.)

Reply:
I think it's appropriate.. Any error makes the connection unresponsive.